frontend/lib/src/WidgetStateManager.ts

```diff
@@ -217,21 +217,18 @@ export class WidgetStateManager {
 
     const submitButtons = this.formsData.submitButtons.get(formId)
 
+    let selectedSubmitButton
+
     if (actualSubmitButton !== undefined) {
-      // Create the button's triggerValue. Just like with a regular button,
-      // `st.form_submit_button()` returns True during a rerun after
-      // it's clicked.
-      this.createWidgetState(actualSubmitButton, {
+      selectedSubmitButton = actualSubmitButton
+    } else if (submitButtons !== undefined && submitButtons.length > 0) {
+      selectedSubmitButton = submitButtons[0]
+    }
+
+    if (selectedSubmitButton) {
+      this.createWidgetState(selectedSubmitButton, {
         fromUi: true,
       }).triggerValue = true
-    } else {
-      // can have an empty list of submitButtons
-      if (submitButtons !== undefined && submitButtons.length > 0) {
-        // click the first submit button. We can choose any so we just choose first.
```

Comment:
Nit: can you include this one comment above in the changes as well

Reply:
Oops! Thank you!